eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1588,7 +1585,47 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
     // peer doesn't cancel the timer
     case Event(TickChannelOpenTimeout, _) => stay()
 
-    // we declare WatchFundingSpentTriggered handlers here because they apply to variants of each state in OFFLINE/SYNCING
+    case Event(w: WatchPublishedTriggered, d: PersistentChannelData) =>
+      log.info(s"zero-conf funding txid=${w.tx.txid} has been published")
+      val fundingStatus = LocalFundingStatus.ZeroconfPublishedFundingTx(w.tx)
+      val metaCommitments1 = d.metaCommitments.updateLocalFundingStatus(w.tx.txid, fundingStatus)
+      val d1 = d match {
+        case d: DATA_WAIT_FOR_FUNDING_CONFIRMED => d.copy(metaCommitments = metaCommitments1)
+        case d: DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED =>
+          val realScidStatus = RealScidStatus.Unknown
+          val shortIds = createShortIds(d.channelId, realScidStatus)
+          // NB: we discard remote's stashed channel_ready, they will send it back at reconnection
+          DATA_WAIT_FOR_DUAL_FUNDING_READY(metaCommitments1, shortIds)
+        case d: DATA_WAIT_FOR_CHANNEL_READY => d.copy(metaCommitments = metaCommitments1)
+        case d: DATA_WAIT_FOR_DUAL_FUNDING_READY => d.copy(metaCommitments = metaCommitments1)
+        case d: DATA_NORMAL => d.copy(metaCommitments = metaCommitments1)
+        case d: DATA_SHUTDOWN => d.copy(metaCommitments = metaCommitments1)
+        case d: DATA_NEGOTIATING => d.copy(metaCommitments = metaCommitments1)
+        case d: DATA_WAIT_FOR_REMOTE_PUBLISH_FUTURE_COMMITMENT => d.copy(metaCommitments = metaCommitments1)
+        case d: DATA_CLOSING => d // there is a dedicated handler in CLOSING state
```

Comment:
There is no handler for `WatchPublishedTriggered` in `CLOSING`...not sure what we should do though

Reply:
We should only update the state. We already put a `WatchFundingConfirmed`, which will be handled in `CLOSING`.